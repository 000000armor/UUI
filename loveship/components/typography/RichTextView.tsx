import style from './RichTextView.module.scss';
import { RichTextView as uuiRichTextView, RichTextViewProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';

export interface RichTextViewMods {
    size?: '12' | '14' | '16';
}

export const RichTextView = withMods<RichTextViewProps, RichTextViewMods>(uuiRichTextView, (mods: RichTextViewMods) => [style.typographyLoveship, style['typography-' + (mods.size || '14')]]);
