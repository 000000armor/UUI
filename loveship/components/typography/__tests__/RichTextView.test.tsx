import React from 'react';
import { renderer } from '@epam/uui-test-utils';
import { RichTextView } from '../RichTextView';

describe('RichTextView', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <RichTextView>
                    <p>Test text</p>
                </RichTextView>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with extra props', () => {
        const tree = renderer
            .create(
                <RichTextView size="12">
                    <p>Test text</p>
                </RichTextView>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
