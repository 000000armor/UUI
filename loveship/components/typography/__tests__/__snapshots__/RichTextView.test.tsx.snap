// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextView should be rendered correctly 1`] = `
<div
  className="typographyLoveship typography-14"
>
  <p>
    Test text
  </p>
</div>
`;

exports[`RichTextView should be rendered correctly with extra props 1`] = `
<div
  className="typographyLoveship typography-12"
>
  <p>
    Test text
  </p>
</div>
`;
